include/seqan3/std/ranges

```diff
@@ -280,4 +295,646 @@ constexpr bool enable_view<type> = static_cast<bool>(seqan3::detail::enable_view
+ * \tparam  sen_t Type of the sentinel; must satisfy std::Sentinel with it_t.
+ * \param[in]  it The iterator on the underlying range.
+ * \param[in] sen The sentinel on the underlying range
+ * \returns  A view of the elements between it_t and sen_t.
+ * \ingroup ranges
+ *
+ * ### View properties
+ *
+ * This view is **source-only**, it can only be at the beginning of a pipe of range transformations.
+ *
+ * | range concepts and reference_t  | `rrng_t` (returned range type)                     |
+ * |---------------------------------|:--------------------------------------------------:|
+ * | std::ranges::InputRange         | *preserved*                                        |
+ * | std::ranges::ForwardRange       | *preserved*                                        |
+ * | std::ranges::BidirectionalRange | *preserved*                                        |
+ * | std::ranges::RandomAccessRange  | *preserved*                                        |
+ * | std::ranges::ContiguousRange    | *preserved*                                        |
+ * |                                 |                                                    |
+ * | std::ranges::ViewableRange      | *guaranteed*                                       |
+ * | std::ranges::View               | *guaranteed*                                       |
+ * | std::ranges::SizedRange         | *preserved*                                        |
+ * | std::ranges::CommonRange        | *preserved*                                        |
+ * | std::ranges::OutputRange        | *preserved*                                        |
+ * | seqan3::const_iterable_concept  | *preserved*                                        |
+ * |                                 |                                                    |
+ * | seqan3::reference_t             | seqan3::value_type_t<it_t>                         |
+ *
+ * Preservation in this table refers to the properties of the iterator/sentinel pair.
+ *
+ * See the \link view view submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### STD module
+ *
+ * This entity will likely be part of C++20 or C++23. It's API will track current proposals and not be stable
+ * within SeqAn3 releases. It is implemented via the range-v3 library or the standard library (if available).
+ * Should it become clear that it will not become part of a future standard, it will migrate to a regular SeqAn3
+ * module.
+ *
+ * ### Example
+ *
+ * \snippet test/snippet/std/view/subrange.cpp example
+ * \hideinitializer
+ */
+template <std::Iterator it_t, std::Sentinel<it_t> sen_t>
+using subrange = ::ranges::subrange<it_t, sen_t>;
+
+} // namespace std::ranges
+
+// -----------------------------------------------------------------------------------------------------
+// View adaptor
+// -----------------------------------------------------------------------------------------------------
+
+namespace std::ranges::view
```

Comment:
I thought `std::view`? .... I don't like namespaces :weary:

Reply:
everything is defined in std::ranges::view and then std::view is declared an alias of std::ranges::view. This enables view stuff to use stuff from std::ranges without prefix, but still appear in a different namespace.